sphinx/source/docs/user_guide/graphs.rst

```diff
@@ -0,0 +1,202 @@
+.. _userguide_graphs:
+
+Visualizing Network Graphs
+==========================
+
+Bokeh has added native support for creating network graph visualizations with
+configurable interactions between edges and nodes.
+
+Edge and Node Renderers
+-----------------------
+
+The secret sauce of the new GraphRenderer is that it maintains separate
+sub-GlyphRenderers for the graph nodes and the graph edges. This allows for
+customizing the nodes by modifying the GraphRenderer's ``node_renderer``
+property. It's even possible to replace the default Circle node glyph with any
+XYGlyph instance, for example a Rect or Oval glyph. Similarly, the style
+properties of the edges can modified through the ``edge_renderer`` property.
+The edge glyph, however, is currently limited to a MultiLine glyph.
+
+There are a couple requirements for the data sources belonging to these
+sub-renderers:
+* The ColumnDataSource associated with the node sub-renderer must have a column
+named ``"index"`` that contains the unique indices of the nodes.
+* The ColumnDataSource associated with the edge sub-renderer has two required
+columns: ``"start"`` and ``"end"``. These columns contain the node indices of
+for the start and end of the edges.
+
+It's possible to add extra meta-data to these data sources to in order to
+add vectorized glyph styling or make data available for callbacks or hover
+tooltips.
+
+Here's a code snippet that:
+
+- replaces the node glyph with an Oval
+- sets the ``height`` and ``width`` attributes of the Oval as scalar values
+- sets the ``fill_color`` attribute of the Oval as a vectorized field and adds the values to the node data source.
+
+.. code-block:: python
+
+    import math
+
+    from bokeh.models import GraphRenderer, Oval
+    from bokeh.palettes import Spectral8
+
+    N = 8
+    node_indices = list(range(N))
+
+    plot = figure(title="Graph Layout Demonstration", x_range=(-1.1,1.1), y_range=(-1.1,1.1),
+                  tools="", toolbar_location=None)
+
+    graph = GraphRenderer()
+
+    graph.node_renderer.glyph = Oval(height=0.1, width=0.2, fill_color="fill_color")
+    graph.node_renderer.data_source.data = dict(
+        index=node_indices,
+        fill_color=Spectral8)
+
+    graph.edge_renderer.data_source.data = dict(
+        start=[0]*N,
+        end=node_indices)
```

Comment:
This should be in a `.py` file in the `examples` dir, then you can use `bokeh-plot` to show the actual result in the docs. The files are named `<chapter>_<desc>.py` so this might be `graph_customize.py`